### ISSUE TITLE
None value for PNG filters 

### ISSUE THREAD

**Issue by omkar-mohanty** [pdf-rs/pdf#180]
While executing the [read.rs](https://github.com/pdf-rs/pdf/blob/master/pdf/examples/read.rs) example, I tried to extract images from a pdf with PNG images in it. The `raw_image_data` function returned `None`  for filter ,  but shouldn't the correct returned filter be a `FlateDecode` ?

**Comment by s3bk** [comment]
Sounds about right.

Could you adjust https://github.com/pdf-rs/pdf/blob/a6e2abc96b23b64aa1051966bb000aabf1275d9f/pdf/src/object/types.rs#L557 accordingly and make a PR?

**Comment by omkar-mohanty** [comment]
Sure :+1: . I had a question though does using the `raw_image_data` on a PNG image return the IDAT chunk?

**Comment by s3bk** [comment]
I don't think so. So you may have to do some reconstruction, and possibly add a as_png method that does this.
